lib/engines/cron/CronJob.ts

```diff
@@ -56,12 +56,9 @@ export class CronJob extends Job {
       this.schedule.cronExpression,
       {
         timezone: this.schedule.timezone,
+        protect: this.preventOverrun
       },
-      () => {
-        if (!this.task.isExecuting || !this.preventOverrun) {
-          this.task.execute()
-        }
-      }
+      async () => await this.task.execute()
```

Comment:
@Hexagon Is it necessary to have an async function here? Wouldn't it suffice to do `return Promise.resolve().then(() => this.task.execute())`?

Reply:
Main reason for the async/await is that croner awaits the call internally, to make `.isBusy()` of the returned job work. Not too familiar with "raw" promises, guess you know better than me what should be used here 👀

Reply:
since you are awaiting it anyway, doing so here makes little sense.
mixing async functions and callbacks can be error-prone, so I would avoid introducing async function without a good reason.
`() => return this.task.execute() might pass the promise further, allowing the croner await to work?`
if croner does not necessarily expect a fully formed promise there (e. g. does not attach .then or .catch handlers to it), that should be sufficient, as `await` will work with both synchronous and asynchronous tasks, yes

Reply:
👍 Will give that a try

Reply:
Hmm, the test only pass using async/await

Reply:
what error are you getting?

Reply:
The overrun protection does not work, the function is triggered again even though it should be blocking for 5000ms (pattern `*/2 * * ..`)

Reply:
Looks like you are depending on it being a function then.
So I would suggest going the Promise.resolve route then

Reply:
I can't seem to get this test working, can you have a look?

Reply:
sure, I'll check it out tomorrow